overcast.py

```diff
@@ -7,51 +7,105 @@
 import requests
 import lxml.html
 import urlparse
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
+log = logging.getLogger('overcast-sonos')
 
 class Overcast(object):
+
     def __init__(self, email, password):
         self.session = requests.session()
         r = self.session.post('https://overcast.fm/login', {'email': email, 'password': password})
         doc = lxml.html.fromstring(r.content)
         alert = doc.cssselect('div.alert')
         if alert:
             raise Exception("Can't login: {}".format(alert[0].text_content().strip()))
-            
+
     def _get_html(self, url):
         return lxml.html.fromstring(self.session.get(url).content)
-            
+
     def get_active_episodes(self):
         doc = self._get_html('https://overcast.fm/podcasts')
         return [
             self.get_episode_detail(cell.attrib['href'])
             for cell in doc.cssselect('a.episodecell')
             if 'href' in cell.attrib
         ]
-        
+
     def get_episode_detail(self, episode_id):
         episode_href = urlparse.urljoin('https://overcast.fm', episode_id)
         doc = self._get_html(episode_href)
+
+        time_elapsed_seconds = int(doc.cssselect('audio#audioplayer')[0].attrib['data-start-time'])
+        time_remaining_seconds = self.get_episode_time_remaining_seconds(episode_id, doc)
+        duration = time_elapsed_seconds + time_remaining_seconds
+        if time_elapsed_seconds == duration:
+            duration = 0
+
         return {
             'id': episode_href.lstrip('/'),
             'title': doc.cssselect('div.titlestack div.title')[0].text_content(),
             'podcast_title': doc.cssselect('div.titlestack div.caption2 a')[0].text_content(),
-            'duration':  60, # fixme - fuck where do I get the duration from :(
+            'offsetMillis': time_elapsed_seconds * 1000,
+            'duration': duration,
+            'data_item_id': doc.cssselect('audio#audioplayer')[0].attrib['data-item-id'],
+            'data_sync_version': doc.cssselect('audio#audioplayer')[0].attrib['data-sync-version'],
+            'albumArtURI': doc.cssselect('div.fullart_container img')[0].attrib['src'],
             'audio_uri': doc.cssselect('audio#audioplayer source')[0].attrib['src'],
             'audio_type': doc.cssselect('audio#audioplayer source')[0].attrib['type'],
         }
-    
+
+    def get_episode_time_remaining_seconds(self, episode_id, episode_html):
```

Comment:
Just awesome that you figured out how to do this. I spent a _long_ time trying to get something like this, to no effect. Yay!